helloworld/helloworld_test.go

```diff
@@ -37,3 +39,27 @@ func Test_Activity(t *testing.T) {
 	require.NoError(t, val.Get(&res))
 	require.Equal(t, "Hello World!", res)
 }
+
+func Test_Using_DevServer(t *testing.T) {
+	testServerEnv := tsenv.New("", "hello-world")
+	err := testServerEnv.Start()
+	require.NoError(t, err)
+
+	w, err := testServerEnv.Worker()
+	require.NoError(t, err)
+	require.NotNil(t, w)
+
+	w.RegisterWorkflow(Workflow)
+	w.RegisterActivity(Activity)
+
+	c, err := testServerEnv.Client()
+	require.NoError(t, err)
+	require.NotNil(t, c)
+
+	result, err := Run(c, testServerEnv.TaskQ)
```

Comment:
For such simple workflow-start code, might as well just do the start from the client in here directly instead of abstracting the simple `ExecuteWorkflow` call as if that part needs testing too. That's just the simplest start-workflow-and-wait, not sure it deserves its own `run.go` file and such, this is a simple hello world. This test can just demonstrate testing the workflow.

Reply:
The intent was to abstract the common code and show the actual workflow code (dev, prod etc) can be tested as is with DevServer. You don't think there's value in that ? I can add some comment to the code if that would highlight the intent.

On the other hand, not married to that either if you think duplicating code in test is ok.

Reply:
That makes sense to show a client can be used, but I think just using the client to start a workflow in the test the same way the starter's main does is good enough to demonstrate that. My fear though is that we're complicating the hello world just to show this abstraction.

Reply:
Sure, I'll move that code inline with just a comment stating this is the same code that runs in starter.

Reply:
Done.